### PR TITLE
Update docs for registry changes 

### PR DESCRIPTION
**Pull request by jmrossy**
- Document registry and update related parts
- Raise explorer section up to top of guides sidebar
- Simplify static img links
- Add PI term to glossary

Fixes https://github.com/hyperlane-xyz/hyperlane-monorepo/issues/2476

### CODE REVIEW

File: docs/partials/deploy-hyperlane/_deploy-contracts.mdx [hunk 1]
```diff
@@ -38,12 +38,11 @@ When asked about the multisig type, choose `message id ism`. In the context of t
 
 ### Dry Run
 
-Prior to executing a deployment, you can execute a dry-run via `--dry-run`, `-dr`, or `-d` to ensure that the deployment will be successful as well as provide analytics on the gas costs of deployment.
-This will not deploy any contracts, but you will be able to see all the contracts that will be deployed, along with the generated artifacts at `$OUT_DIR/dry-run_*`.
+Prior to executing a deployment, you can execute a dry-run via `--dry-run <CHAIN_NAME>`  or `-d <CHAIN_NAME>` to ensure that the deployment will be successful as well as provide analytics on the gas costs of deployment.
+This will not deploy any contracts, but you a list of all the deployed contracts that will be will be displayed.
```
Comment (nbayindirli) [inline docs/partials/deploy-hyperlane/_deploy-contracts.mdx:42]: typo in `, but you a list of all`

File: docs/partials/deploy-hyperlane/_deploy-contracts.mdx [hunk 2]
```diff
@@ -70,15 +68,14 @@ We're now ready to use the `deploy core` command to deploy the Hyperlane contrac
...
 ```
 
 ### Verify
 
-Deployment contract artifacts will be written to to the `artifacts/` folder by default. The deployer will create two timestamped files, `agent-config-{timestamp}.json` and `core-deployment-{timestamp}.json`
-The `core-deployment` file contains the addresses of core contracts, organized by chain.
-The `agent-config` file contains the data needed to run Hyperlane agents for the next step.
+Deployment contract artifacts will be written to the configured registries. By default, you'll find the new addresses in the local directory you ran the command from (e.g. `./chains/chain1/addresses.yaml`).
+
+The deployment will also generate a Hyperlane Agent config, which is written to `./configs/agent.json` by default but can be configured with the `--agent` flag.
```
Comment (nbayindirli) [inline docs/partials/deploy-hyperlane/_deploy-contracts.mdx:81]: nit: comma in `by default, but`

File: src/components/MessageQuickstart.tsx [hunk 3]
```diff
@@ -33,7 +33,8 @@ export default function MessageQuickstart() {
             </CodeBlock>
             Send the message:
             <CodeBlock language="shell">
-              hyperlane send message --origin {originChain} --destination{" "}
+              hyperlane send message --origin {originChain} --destination --body
```
Comment (yorhodes) [inline src/components/MessageQuickstart.tsx:36]: seems like destination is missing now? also does this newline work when copy pasted?

File: package.json [hunk 4]
```diff
@@ -21,7 +21,7 @@
     "@docusaurus/theme-mermaid": "3.0.1",
     "@easyops-cn/docusaurus-search-local": "^0.37.4",
     "@hyperlane-xyz/core": "^3.10.0",
```
Comment (yorhodes) [inline package.json:23]: should we bump this?

File: docs/deploy-hyperlane-troubleshooting.mdx [hunk 5]
```diff
@@ -1,8 +1,16 @@
 # Troubleshooting
 
+## Logging
+
+Logging levels and formats can be configured for Hyperlane tooling using the following two environment variables:
+- `LOG_LEVEL`: The log level to filter to. Defaults to `info`. Choices: `debug | info | warn | error | off`
+- `LOG_FORMAT`: The format of the log output. Defaults to `pretty`. Choices: `pretty | json`
```
Comment (yorhodes) [inline docs/deploy-hyperlane-troubleshooting.mdx:7]: I think json was breaking for me fwiw

File: docs/guides/deploy-warp-route.mdx [hunk 6]
```diff
@@ -21,8 +23,8 @@ Check out the [Warp Route](/docs/reference/applications/warp-routes) reference p
 
 To deploy the route, you will need a Warp Route deployment config file. A valid config will specify:
 * Which token, on which chain, is this Warp Route being created for?
-* _Optional:_ Hyperlane connection details including contract addresses for the [mailbox](../reference/messaging/messaging-interface.mdx), [interchain gas](../reference/hooks/interchain-gas.mdx), and [interchain security modules](../reference/ISM/specify-your-ISM.mdx).
-* _Optional:_ The token standard - fungible tokens using ERC20 or NFTs using ERC721. Defaults to ERC20.
+* _Optional:_ Hyperlane connection details including contract addresses for the [mailbox](/docs/reference/messaging/messaging-interface.mdx), [interchain gas](/docs/reference/hooks/interchain-gas.mdx), and [interchain security modules](/docs/reference/ISM/specify-your-ISM.mdx).
+* _Optional:_ The token standard - fungible tokens using ERC20 or NFTs using ERC721. Note that ERC721 support is experimental and some Hyperlane tooling won't work for NFTs yet.
```
Comment (yorhodes) [inline docs/guides/deploy-warp-route.mdx:27]: 🙏

File: docs/guides/deploy-warp-route.mdx [hunk 7]
```diff
@@ -37,57 +39,36 @@ Or see below for details on how to define your config manually.
...
+Your config consist of a map of chain names to deployment configs. Each config sets details about the token for which you are creating a Warp Route.
 
-* **chainName**: Set this equal to the chain on which your token exists
 * **type**:
-    - Set this to `TokenType.collateral` to create a basic Warp Route for an ERC20/ERC721 token.
-    - Set this to `TokenType.collateralVault` to create a yield-bearing Warp Route for an ERC20 token that deposits into an ERC4626 vault.
-    - Set this to `TokenType.native` to create a Warp Route for a native token (e.g. ether)
+    - Set this to `collateral` to create a basic Warp Route for an ERC20/ERC721 token.
+    - Set this to `collateralVault` to create a yield-bearing Warp Route for an ERC20 token that deposits into an ERC4626 vault.
+    - Set this to `native` to create a Warp Route for a native token (e.g. ether)
```
Comment (yorhodes) [inline docs/guides/deploy-warp-route.mdx:49]: 🙏

File: docs/guides/deploy-warp-route.mdx [hunk 8]
```diff
@@ -37,57 +39,36 @@ Or see below for details on how to define your config manually.
...
-    - If using `TokenType.collateralVault`, the address of the ERC4626 vault to deposit collateral into
-* **isNft:** If using `TokenType.collateral` for an ERC721 contract, set to `true`.
-
-#### Synthetics
-
-Your config must have at least one `synthetics` entry. Here you will configure details about the remote chains supported by your Warp Route.
-
-* **chainName:** Set this equal to the chain on which you want a wrapped version of your token
+    - If using `collateral`, the address of the ERC20/ERC721 contract for which to create a route
+    - If using `collateralVault`, the address of the ERC4626 vault to deposit collateral into
```
Comment (yorhodes) [inline docs/guides/deploy-warp-route.mdx:52]: we could maybe add some stuff around XERC20 or FiatToken here cc @avious00

File: docs/reference/cli.mdx [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+To view a list of the available commands and their arguments, run `hyperlane --help`.
+
+A few common command options include:
+- `--version`: Display your current installed version of the CLI
+- `--registry`: Set the registry URI to use for chain metadata and contract addresses
+- `--overrides`: Set an additional path for overrides to the default registry data
+-- `log`: Set the log format for the CLI (defaults to pretty)
+-- `verbosity`: Set the log level for the CLI (defaults to info)
+-- `key`: Set the key to use for signing transactions (defaults to the HYP_KEY env var)
```
Comment (yorhodes) [inline docs/reference/cli.mdx:45]: why format these differently?

File: docs/reference/registries.mdx [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+
+As long as the registry conforms to the same layout and file schemas as the default registry, any Github repository URL or local file path can be used instead.
+
+The CLI has two command line arguments for configuring the registry it uses:
+- The `--registry` flag sets the primary registry and uses the default Github registry if not set
+- The `--overrides` flag sets an optional additional path which can be useful for forcing specific overrides of the default data, for example to use a different RPC URL in a chain's metadada.
+
+For more information see the [CLI reference page](/docs/reference/cli.mdx).
+
+## Usage (Javascript)
```
Comment (yorhodes) [inline docs/reference/registries.mdx:28]: or typescript?

File: docs/reference/registries.mdx [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+
+However, community members are encouraged to add their own deployments to this default registry. The steps to contribute are as follows:
+1. Fork and clone the registry repository
+2. Use the CLI and set the `--registry` flag to the local path of the cloned repository
+3. Execute your commands as needed
+4. Push your changes to your fork and create a pull request
+
+## Custom registries
+
+As long as the registry conforms to the same layout and file schemas as the default registry, any Github repository URL or local file path can be used instead.
```
Comment (yorhodes) [inline docs/reference/registries.mdx:20]: maybe add a note about enabling testing with the fork / PR before merging into the default registry

File: docs/reference/registries.mdx [hunk 12]
```diff
@@ -0,0 +1,61 @@
+# Hyperlane Registries
+
+## Overview
+
+Hyperlane can be deployed to anywhere by anyone. But for those deployments to be useful, their details must be well known. Registries are collections of chain metadata, contract addresses, and other useful information for interacting with Hyperlane contracts and tools. There is a [default registry](https://github.com/hyperlane-xyz/hyperlane-registry) maintained by the Hyperlane core team. The [CLI](/docs/reference/cli.mdx), the [Explorer](https://explorer.hyperlane.xyz), and other tools can use registry data. 
+
+## Default registry
```
Comment (yorhodes) [inline docs/reference/registries.mdx:7]: maybe "Canonical"?

**Review by yorhodes — APPROVED** [review]

File: docs/reference/registries.mdx (hunk 12 above)
Comment (jmrossy) [reply docs/reference/registries.mdx:7]: Agreed

**Pull request merged**